### ISSUE TITLE
Production deployments failing

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/perennial#88]
Just tried to deploy the area-model sims (uses newer chipper SHAs), and they failed like:
```
Apr 24 16:24:34 phet-server.int.colorado.edu build-server[76757]: Building brand: phet
Apr 24 16:24:36 phet-server.int.colorado.edu build-server[76757]: debug: stdout: >> require.js optimization for brand: phet complete (4171537 bytes)
Apr 24 16:24:57 phet-server.int.colorado.edu build-server[76757]: debug: stdout: >> Minification for phet complete
Apr 24 16:24:57 phet-server.int.colorado.edu build-server[76757]: debug: stdout: >> Require.js: 1312897 bytes
Apr 24 16:24:57 phet-server.int.colorado.edu build-server[76757]: >> Preloads: 184819 bytes
Apr 24 16:24:57 phet-server.int.colorado.edu build-server[76757]: >> Mipmaps: 770828 bytes
Apr 24 16:24:59 phet-server.int.colorado.edu build-server[76757]: debug: stdout:
Apr 24 16:24:59 phet-server.int.colorado.edu build-server[76757]: Done.
Apr 24 16:24:59 phet-server.int.colorado.edu build-server[76757]: debug: deploying to servers: ["dev","production"]
Apr 24 16:24:59 phet-server.int.colorado.edu build-server[76757]: info: deploying to dev
Apr 24 16:24:59 phet-server.int.colorado.edu build-server[76757]: info: running mkdir -p /data/web/htdocs/dev/html/area-model-algebra/1.0.0 remotely on bayes.colorado.edu
Apr 24 16:24:59 phet-server.int.colorado.edu build-server[76757]: debug: running ssh phet-admin@bayes.colorado.edu mkdir -p /data/web/htdocs/dev/html/area-model-algebra/1.0.0 from .
Apr 24 16:25:00 phet-server.int.colorado.edu build-server[76757]: info: Writing file to path: ../area-model-algebra/build/phet/.rsync-filter
Apr 24 16:25:01 phet-server.int.colorado.edu build-server[76757]: debug: Write success.
Apr 24 16:25:01 phet-server.int.colorado.edu build-server[76757]: info: running chmod -R g+w /data/web/htdocs/dev/html/area-model-algebra/1.0.0 remotely on bayes.colorado.edu
Apr 24 16:25:01 phet-server.int.colorado.edu build-server[76757]: debug: running ssh phet-admin@bayes.colorado.edu chmod -R g+w /data/web/htdocs/dev/html/area-model-algebra/1.0.0 from .
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: info: deploying to production
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: info: deploying brand: phet
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: running mv area-model-algebra_all_iframe_phet.html area-model-algebra_all_iframe.html from ../area-model-algebra/build/phet
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: running mv area-model-algebra_all_phet.html area-model-algebra_all.html from ../area-model-algebra/build/phet
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: running mv area-model-algebra_all_phet_debug.html area-model-algebra_all_debug.html from ../area-model-algebra/build/phet
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: running mv area-model-algebra_en_iframe_phet.html area-model-algebra_en_iframe.html from ../area-model-algebra/build/phet
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: running mv area-model-algebra_en_phet.html area-model-algebra_en.html from ../area-model-algebra/build/phet
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: info: Error on mkdir, probably due to the fact that the target already exists
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: Copying recursive ../area-model-algebra/build/phet to /data/web/static/phetsims/sims/html/area-model-algebra/1.0.0/
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: rsync: mkdir "/data/web/static/phetsims/sims/html/area-model-algebra/1.0.0" failed: No such file or directory (2)
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: rsync error: error in file IO (code 11) at main.c(587) [Receiver=3.0.9]
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: rsync: connection unexpectedly closed (9 bytes received so far) [sender]
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: rsync error: error in rsync protocol data stream (code 12) at io.c(605) [sender=3.0.9]
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: 12
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: rsync -razp --exclude=.rsync-filter ../area-model-algebra/build/phet/ /data/web/static/phetsims/sims/html/area-model-algebra/1.0.0/
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: error: BUILD ABORTED! Error: rsync exited with code 12
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: info: build aborted: checking out master for every repo in case build shas are still checked out
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: running grunt checkout-master-all from .
Apr 24 16:25:02 phet-server.int.colorado.edu build-server[76757]: debug: stdout: Running "checkout-master-all" task
Apr 24 16:25:03 phet-server.int.colorado.edu build-server[76757]: debug: stdout:
Apr 24 16:25:03 phet-server.int.colorado.edu build-server[76757]: debug: stdout: Done.
Apr 24 16:25:03 phet-server.int.colorado.edu build-server[76757]: error: Build failure: Build aborted,Error: rsync exited with code 12. Sim = area-model-algebra Version = 1.0.0 Locales = * Shas = {"comment":"# area-model-algebra 1.0.0 Tue Apr 24 2018 16:23:42 GMT-0600 (MDT)","a
Apr 24 16:25:03 phet-server.int.colorado.edu build-server[76757]: D"},"vibe":{"sha":"709fbf211ab1d9cb977921a0b521fa2348f80fe3","branch":"HEAD"}}
Apr 24 16:25:03 phet-server.int.colorado.edu build-server[76757]: info: attempting to send email
```

@mattpen or @jbphet, thoughts?

**Comment by jonathanolson** [comment]
Looks tentatively like it's trying to create `/data/web/static/phetsims/sims/html/area-model-algebra/1.0.0/` without first creating `/data/web/static/phetsims/sims/html/area-model-algebra/`. Then it looks like it's hardcoded to ignore any error in this operation, and the rsync recursive copies then fail out.

**Comment by mattpen** [comment]
> Looks tentatively like it's trying to create /data/web/static/phetsims/sims/html/area-model-algebra/1.0.0/ without first creating /data/web/static/phetsims/sims/html/area-model-algebra/.

It should be creating `/data/web/static/phetsims/sims/html/area-model-algebra/1.0.0/` with the flag to create any necessary parent directories.  I'll verify.

**Comment by mattpen** [comment]
It looks like I made some incorrect assumptions about how `fs.mkdir` was handling missing parents.  I added checks to make sure the sim directory was created before attempting to create the version directory, as well as adding checks for allowing `EEXIST` errors to occur (these are expected and should be allowed to proceed).  I deployed the changes to phet-server and @jonathanolson successfully built area-model-introduction with this change.
